brave/brave-core
# Revert "Remove navigator.deviceMemory (part 1)"

Fixes brave/brave-browser#12289

This reverts commit 740b4e315cdebb92aa9c022bfcda357ea13f84ea which was added in https://github.com/brave/brave-core/pull/3669 to fix #1157.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

Testing is not 100% reliable, but the idea is to compare a current version of Brave with this build:

1. Using Nightly (without this PR), clear all cookies, history, etc.
2. Open the devtool console and type `navigator.deviceMemory` (should return `undefined`).
3. Log into Paypal and suffer through the CAPTCHA.
4. Using this build, clear all cookies, history, etc.
5. Open the devtool console and type `navigator.deviceMemory` (should return a number).
6. Log into Paypal and you shouldn't get a CAPTCHA.
7. Open https://antoinevastel.com/bots/ and confirm that everything is green at the bottom:
![Screenshot from 2020-10-27 12-15-34](https://user-images.githubusercontent.com/167821/97350798-23149800-184e-11eb-8934-dcba6e744672.png)
8. Open https://arh.antoinevastel.com/bots/areyouheadless and confirm that it says "You are not Chrome headless".

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
